src/pipes/db-lookup.pipe.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from "class-validator";
+import { FindOneOptions } from "typeorm";
+import { DbLookupValidator } from '../validators';
+
+/**
+ * Configuration interface for validation-specific options
+ * Used by the DbLookup decorator
+ */
+
+export interface DbLookupConfig {
+  validateNumeric?: boolean;
+  ignoreEmpty?: boolean;
+}
+
+/**
+ * Database lookup validation decorator
+ * @param type - The entity type to validate against
+ * @param findOption - Function to generate FindOneOptions or 'primary' for primary key lookup
+ * @param validationOptions - Standard class-validator options
+ * @param dbLookupConfig - Configuration options for the lookup
+ */
+export function DbLookup(
+  type: any,
+  findOption: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary",
+  validationOptions: ValidationOptions,
+  dbLookupConfig: DbLookupConfig = {}
+) {
+  return function (object: Object, propertyName: string) {
+    registerDecorator({
+      name: "dbLookup",
+      target: object.constructor,
+      propertyName: propertyName,
+      options: validationOptions,
+      constraints: [{ ignoreEmpty: dbLookupConfig.ignoreEmpty ?? true, type, findOption }],
```

Comment:
The `validateNumeric` argument should be passed in here:
```suggestion
      constraints: [
        {
          ignoreEmpty: dbLookupConfig.ignoreEmpty ?? true,
          validateNumeric: dbLookupConfig.validateNumeric ?? false,
          type,
          findOption,
        },
      ],
```